hadoop/src/main/java/org/apache/carbondata/hadoop/stream/StreamRecordReader.java

```diff
@@ -387,7 +392,12 @@ protected void readRowFromStream() {
         }
       } else {
         if (isNoDictColumn[colCount]) {
-          int v = input.readShort();
+          int v = 0;
+          if (dimensionDataTypes[colCount] == DataTypes.VARCHAR) {
```

Comment:
better to use a boolean array to store the result.

Reply:
Done